### PR TITLE
Add [Values] attribute support in test browser

### PR DESCRIPTION
**Pull request by peppy**
Note that the test is only really useful for visually confirming this behaviour. Hard to really do any other testing.

### CODE REVIEW

File: osu.Framework/Testing/TestBrowser.cs [hunk 1]
```diff
@@ -651,4 +682,53 @@ public enum TestBrowserAction
         Reload,
         Search
     }
+
+    public class CartesianProduct<T>
```
Comment (frenzibyte) [inline osu.Framework/Testing/TestBrowser.cs:686]: Leftover class?

Comment (peppy) [reply osu.Framework/Testing/TestBrowser.cs]: Yep, removed.

File: osu.Framework/Testing/TestBrowser.cs [hunk 2]
```diff
@@ -504,12 +504,43 @@ private void finishLoad(TestScene newTest, Action onCompletion)
                 {
                     string repeatSuffix = i > 0 ? $" ({i + 1})" : string.Empty;
 
-                    if (m.GetCustomAttribute(typeof(TestAttribute), false) != null)
+                    var methodWrapper = new MethodWrapper(m.GetType(), m);
+
+                    if (methodWrapper.GetCustomAttributes<TestAttribute>(false).FirstOrDefault() != null)
```
Comment (bdach) [inline osu.Framework/Testing/TestBrowser.cs:509]: typical nit - I'd either use

```suggestion
                    if (methodWrapper.GetCustomAttributes<TestAttribute>(false).SingleOrDefault() != null)
```

(`AllowMultiple = false` on `TestAttribute` allows this), or use `Any()`.

File: osu.Framework/Testing/TestBrowser.cs [hunk 3]
```diff
@@ -504,12 +504,43 @@ private void finishLoad(TestScene newTest, Action onCompletion)
...
-                        CurrentTest.AddLabel($"{name}{repeatSuffix}");
+                        var parameters = m.GetParameters();
+
+                        if (parameters.Length > 0)
+                        {
+                            var valueMatrix = new List<List<object>>();
+
+                            foreach (var p in methodWrapper.GetParameters())
+                            {
+                                var valueAttrib = p.GetCustomAttributes<ValuesAttribute>(false).FirstOrDefault();
```
Comment (bdach) [inline osu.Framework/Testing/TestBrowser.cs:519]: same here

```suggestion
                                var valueAttrib = p.GetCustomAttributes<ValuesAttribute>(false).SingleOrDefault();
```

**Pull request merged**